src/main/java/com/minecolonies/coremod/entity/citizenhandlers/CitizenHappinessHandler.java

```diff
@@ -0,0 +1,596 @@
+            {
+                chatSpamFilter.talkWithoutSpam("entity.citizen.noHouse", citizen.getName());
+            }
+        }
+        else
+        {
+            numberOfDaysWithoutHouse = 0;
+        }
+        setHomeModifier(hasHouse);
+
+        if (!hasJob)
+        {
+            numberOfDaysWithoutJob++;
+            if (numberOfDaysWithoutJob > DEMANDS_DAYS_WITHOUT_JOB)
+            {
+                chatSpamFilter.talkWithoutSpam("entity.citizen.demandsJob", citizen.getName());
+            }
+            else if (numberOfDaysWithoutJob > COMPLAIN_DAYS_WITHOUT_JOB)
+            {
+                chatSpamFilter.talkWithoutSpam("entity.citizen.noJob", citizen.getName());
+            }
+        }
+        else
+        {
+            numberOfDaysWithoutJob = 0;
+        }
+        setJobModifier(hasJob);
+
+        farmerModifier = 0;
+        hasNoFields = true;
+        for (final FieldDataModifier field : fieldModifier.values())
+        {
+            if (field.isCanFarm())
+            {
+                farmerModifier += FIELD_MODIFIER_POSITIVE;
+                hasNoFields = false;
+            }
+            else
+            {
+                field.increaseInactiveDays();
+                if (field.getInactiveDays() < NO_FIELDS_COMPLAINS_DAYS)
+                {
+                    farmerModifier += FIELD_MODIFIER_MIN;
+                }
+                else
+                {
+                    farmerModifier += ((double) ((double) field.getInactiveDays() / FIELD_MAX_DAYS_MODIFIER) * FIELD_MODIFIER_MAX) + FIELD_MODIFIER_MIN;
+                }
+            }
+        }
+
+        if (hasNoFields)
+        {
+            farmerModifier = NO_FIELD_MODIFIER;
+        }
+
+        noToolModifier = 0;
+        for (final Map.Entry<IToolType, Integer> entry : needsTool.entrySet())
+        {
+            final int numDays = entry.getValue() + 1;
+            final IToolType toolType = entry.getKey();
+            needsTool.put(toolType, numDays);
+            if (numDays > NO_TOOLS_DEMANDS_DAYS)
+            {
+                chatSpamFilter.talkWithoutSpam("entity.citizen.noTool", citizen.getName(), toolType.getDisplayName());
+            }
+            else if (numDays > NO_TOOLS_COMPLAINS_DAYS)
+            {
+                chatSpamFilter.talkWithoutSpam("entity.citizen.demandsTool", citizen.getName(), toolType.getDisplayName());
+            }
+            noToolModifier += ((double) ((double) numDays / NO_TOOLS_MAX_DAYS_MODIFIER) * NO_TOOLS_MODIFIER);
+        }
+
+        citizen.markDirty();
+
+    }
+
+    /**
+     * this function applies adjust to happiness.
+     * This would mean the citizen is full.
+     */
+    public void setSaturated()
+    {
+        foodModifier = FOOD_MODIFIER_POSITIVE;
+    }
+
+    /**
+     * set the Damage modifier on the citizens happiness
+     * depending on how hurt they are.
+     */
+    public void setDamageModifier()
```

Comment:
### Errors:
- 'public' method setDamageModifier() is not exposed via an interface